src/components/PointRanking.js

```diff
@@ -4,29 +4,96 @@ import {connect} from "react-redux";
+                    this.setState({data : cons.sort((a, b) => b.rider.startNr - a.rider.startNr), sortOrder: 'ascending', uiOrder:'descending'});
+                break;
+            case 'name':
+                this.state.sortOrder === 'ascending' ? this.setState({data : cons.sort((a, b) => a.rider.name.replace(/ /g,'').toLowerCase().localeCompare(b.rider.name.replace(/ /g,'').toLowerCase())), sortOrder: 'descending', uiOrder :'ascending'}) :
+                    this.setState({data : cons.sort((a, b) => b.rider.name.replace(/ /g,'').toLowerCase().localeCompare(a.rider.name.replace(/ /g,'').toLowerCase())), sortOrder: 'ascending', uiOrder:'descending'});
+                break;
+            case 'team':
+                this.state.sortOrder === 'ascending' ? this.setState({data : cons.sort((a, b) => a.rider.teamShortName.localeCompare(b.rider.teamShortName)), sortOrder: 'descending', uiOrder :'ascending'}) :
+                    this.setState({data : cons.sort((a, b) => b.rider.teamShortName.localeCompare(a.rider.teamShortName)), sortOrder: 'ascending', uiOrder:'descending'});
+                break;
+            case 'country':
+                this.state.sortOrder === 'ascending' ? this.setState({data : cons.sort((a, b) => a.rider.country.localeCompare(b.rider.country)), sortOrder: 'descending', uiOrder :'ascending'}) :
+                    this.setState({data : cons.sort((a, b) => b.rider.country.localeCompare(a.rider.country)), sortOrder: 'ascending', uiOrder:'descending'});
+                break;
+            default: // Default sorted by rank
+                this.state.sortOrder === 'ascending' ? this.setState({data : cons.sort((a, b) => a.virtualGap - b.virtualGap), sortOrder: 'descending', uiOrder :'ascending'}) :
+                    this.setState({data : cons.sort((a, b) => b.virtualGap - a.virtualGap), sortOrder: 'ascending', uiOrder:'descending'});
+                break;
+        }
+    };
+
+    componentWillReceiveProps(nextProps){
+        const {cons} = nextProps;
+        this.setState({data : cons.sort((a, b) => a.virtualGap - b.virtualGap), sortOrder: 'ascending'});
+        var hashtable = {};
+        cons.sort((a,b) => a.virtualGap - b.virtualGap).map(con => hashtable[con.id] = cons.findIndex(c => c.id === con.id)+1);
+        this.setState({ranking: hashtable});
+    }
+
+    componentDidMount(){
+        const {cons} = this.props;
+        this.setState({data : cons.sort((a, b) => a.virtualGap - b.virtualGap), sortOrder: 'ascending'});
+        var hashtable = {};
+        cons.sort((a,b) => a.virtualGap - b.virtualGap).map(con => hashtable[con.id] = cons.findIndex(c => c.id === con.id)+1);
+        this.setState({ranking: hashtable});
+    }
+
+
+    dataNotReady(){
+        return (
+            <Table.Row>
+                <Table.Cell>Klassemente werden geladen</Table.Cell>
+            </Table.Row>
+        )
+    }
+
     render() {
         const {cons} = this.props;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "cons" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'cons' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)